sqlx-core/src/caching_connection.rs

```diff
@@ -0,0 +1,13 @@
+use futures_core::future::BoxFuture;
+
+use crate::error::Error;
+
+/// A connection that is capable of caching prepared statements.
+pub trait CachingConnection: Send {
```

Comment:
I find these quite useful also from outside.

Reply:
I agree with having the methods. I think I'd rather have them directly on `Connection` like this:

```rust
/* auto */ trait HasStatementCache { }

trait Connection {
  fn statement_cache_size(&self) -> usize 
  where 
    Self::Database: HasStatementCache 
  {
    0
  }
}
```

This removes the extra trait that users would have to import and still blocks using the methods if the database driver doesn't flag itself as having a statement cache.

Thoughts?

Reply:
This is fine for me. I was maybe overengineering this a bit, adding them to the `Connection`.